src/Nethermind/Nethermind.Synchronization/FastSync/DetailedProgress.cs

```diff
@@ -160,5 +166,11 @@ public byte[] Serialize()
 
             return rlp.Bytes;
         }
+
+        public void SetSnapSyncData(bool stateRangesFinished, long stateRangesSyncedBytes)
+        {
+            _snapSyncFinished = stateRangesFinished;
+            Interlocked.CompareExchange(ref DataSize, stateRangesSyncedBytes, 0L);
```

Comment:
I think you can't directly translate snap sync bytes to whole trie bytes. We would have to count data stored in the trie instead of data downloaded from network?

Reply:
That is true, but I think this corresponds better to the estimated size (used to calculate progress in `DetailedProgress` class). For Goerli, this gives ~12.5 GB downloaded in 1st phase of Snap Sync, while the saved tree is almost 30GB. Estimated size is ~13.2 GB.

Reply:
Calculating the exact amont of data that has been added to the trie may require some further work. Current results on Goerli seem to be quite good - I'll test on mainnet.